Abort crossbuild on error

The crossbuild command doesn't seem to get stopped on error

```
promu crossbuild
> running the main builder docker image
Error: Unknown option: linux/amd64 linux/386 darwin/amd64 darwin/386 windows/amd64 windows/386 freebsd/amd64 freebsd/386 openbsd/amd64 openbsd/386 netbsd/amd64 netbsd/386 dragonfly/amd64
Usage: builder.sh [args]
  -i,--import-path arg  : Go import path of the project
  -p,--platforms arg    : List of platforms (GOOS/GOARCH) to build separated by a space
  -T,--tests            : Go run tests then exit
panic: Error: Unknown option: linux/amd64 linux/386 darwin/amd64 darwin/386 windows/amd64 windows/386 freebsd/amd64 freebsd/386 openbsd/amd64 openbsd/386 netbsd/amd64 netbsd/386 dragonfly/amd64
Usage: builder.sh [args]
  -i,--import-path arg  : Go import path of the project
  -p,--platforms arg    : List of platforms (GOOS/GOARCH) to build separated by a space
 [recovered]            : Go run tests then exit
        panic: runtime error: index out of range

goroutine 1 [running]:
panic(0x752d80, 0xc82000e120)
        /usr/lib/go/src/runtime/panic.go:464 +0x3e6
github.com/prometheus/promu/vendor/github.com/progrium/go-shell.(*Process).Error(0xc820134450, 0x0, 0x0)
        /home/grobie/code/go/src/github.com/prometheus/promu/vendor/github.com/progrium/go-shell/shell.go:206 +0x25d
github.com/prometheus/promu/vendor/github.com/progrium/go-shell.ErrExit()
        /home/grobie/code/go/src/github.com/prometheus/promu/vendor/github.com/progrium/go-shell/shell.go:51 +0x181
panic(0x77f2e0, 0xc820134450)
        /usr/lib/go/src/runtime/panic.go:426 +0x4e9
github.com/prometheus/promu/vendor/github.com/progrium/go-shell.(*Command).Run(0xc82008fec0, 0x6)
        /home/grobie/code/go/src/github.com/prometheus/promu/vendor/github.com/progrium/go-shell/shell.go:173 +0x81a
github.com/prometheus/promu/vendor/github.com/progrium/go-shell.Run(0xc82006cfc0, 0x6, 0x6, 0x6c68c0)
        /home/grobie/code/go/src/github.com/prometheus/promu/vendor/github.com/progrium/go-shell/shell.go:218 +0x43
github.com/prometheus/promu/cmd.runCrossbuild()
        /home/grobie/code/go/src/github.com/prometheus/promu/cmd/crossbuild.go:123 +0xd08
github.com/prometheus/promu/cmd.glob.func3(0x99f4c0, 0x9c09b0, 0x0, 0x0)
        /home/grobie/code/go/src/github.com/prometheus/promu/cmd/crossbuild.go:54 +0x14
github.com/prometheus/promu/vendor/github.com/spf13/cobra.(*Command).execute(0x99f4c0, 0x9c09b0, 0x0, 0x0, 0x0, 0x0)
        /home/grobie/code/go/src/github.com/prometheus/promu/vendor/github.com/spf13/cobra/command.go:569 +0x85a
github.com/prometheus/promu/vendor/github.com/spf13/cobra.(*Command).ExecuteC(0x99fac0, 0x99f4c0, 0x0, 0x0)
        /home/grobie/code/go/src/github.com/prometheus/promu/vendor/github.com/spf13/cobra/command.go:656 +0x55c
github.com/prometheus/promu/vendor/github.com/spf13/cobra.(*Command).Execute(0x99fac0, 0x0, 0x0)
        /home/grobie/code/go/src/github.com/prometheus/promu/vendor/github.com/spf13/cobra/command.go:615 +0x2d
github.com/prometheus/promu/cmd.Execute()
        /home/grobie/code/go/src/github.com/prometheus/promu/cmd/promu.go:57 +0x27
main.main()
        /home/grobie/code/go/src/github.com/prometheus/promu/main.go:20 +0x14
```

And another one

```
# darwin-amd64
rm -rf node_exporter node_exporter-0.12.0.darwin-amd64.tar.gz .build *-stamp
cd /go/src/github.com/prometheus/node_exporter && GO15VENDOREXPERIMENT=1 GOPATH=/go /usr/local/go/bin/go build -ldflags "-X main.Version=0.12.0 -X "github.com/prometheus/node_exporter"/collector.Version=0.12.0 -X "github.com/prometheus/node_exporter"/collector.Revision=dd76a68 -X "github.com/prometheus/node_exporter"/collector.Branch=master" -o node_exporter
# github.com/prometheus/node_exporter
/usr/local/go/pkg/tool/linux_amd64/link: /usr/local/go/pkg/tool/linux_amd64/link: combining dwarf failed: open /tmp/go-link-233789061/go.dwarf: no such file or directory
Makefile.COMMON:85: recipe for target 'node_exporter' failed
 [recovered]ode_exporter] Error 2
        panic: runtime error: index out of range

goroutine 1 [running]:
panic(0x752d80, 0xc82000e120)
        /usr/lib/go/src/runtime/panic.go:464 +0x3e6
github.com/prometheus/promu/vendor/github.com/progrium/go-shell.(*Process).Error(0xc8200f6d50, 0x0, 0x0)
        /home/grobie/code/go/src/github.com/prometheus/promu/vendor/github.com/progrium/go-shell/shell.go:206 +0x25d
github.com/prometheus/promu/vendor/github.com/progrium/go-shell.ErrExit()
        /home/grobie/code/go/src/github.com/prometheus/promu/vendor/github.com/progrium/go-shell/shell.go:51 +0x181
panic(0x77f2e0, 0xc8200f6d50)
        /usr/lib/go/src/runtime/panic.go:426 +0x4e9
github.com/prometheus/promu/vendor/github.com/progrium/go-shell.(*Command).Run(0xc820116880, 0x6)
        /home/grobie/code/go/src/github.com/prometheus/promu/vendor/github.com/progrium/go-shell/shell.go:173 +0x81a
github.com/prometheus/promu/vendor/github.com/progrium/go-shell.Run(0xc82012cae0, 0x6, 0x6, 0x6c68c0)
        /home/grobie/code/go/src/github.com/prometheus/promu/vendor/github.com/progrium/go-shell/shell.go:218 +0x43
github.com/prometheus/promu/cmd.runCrossbuild()
        /home/grobie/code/go/src/github.com/prometheus/promu/cmd/crossbuild.go:123 +0xcf0
github.com/prometheus/promu/cmd.glob.func3(0x99f4c0, 0xc820118240, 0x0, 0x1)
        /home/grobie/code/go/src/github.com/prometheus/promu/cmd/crossbuild.go:54 +0x14
github.com/prometheus/promu/vendor/github.com/spf13/cobra.(*Command).execute(0x99f4c0, 0xc820118220, 0x1, 0x1, 0x0, 0x0)
        /home/grobie/code/go/src/github.com/prometheus/promu/vendor/github.com/spf13/cobra/command.go:569 +0x85a
github.com/prometheus/promu/vendor/github.com/spf13/cobra.(*Command).ExecuteC(0x99fac0, 0x99f4c0, 0x0, 0x0)
        /home/grobie/code/go/src/github.com/prometheus/promu/vendor/github.com/spf13/cobra/command.go:656 +0x55c
github.com/prometheus/promu/vendor/github.com/spf13/cobra.(*Command).Execute(0x99fac0, 0x0, 0x0)
        /home/grobie/code/go/src/github.com/prometheus/promu/vendor/github.com/spf13/cobra/command.go:615 +0x2d
github.com/prometheus/promu/cmd.Execute()
        /home/grobie/code/go/src/github.com/prometheus/promu/cmd/promu.go:57 +0x27
main.main()
        /home/grobie/code/go/src/github.com/prometheus/promu/main.go:20 +0x14
```

--- a/cmd/crossbuild.go
+++ b/cmd/crossbuild.go
@@ -125,27 +125,31 @@ func runCrossbuild() {
 
 	if mainPlatformsParam := strings.Join(mainPlatforms[:], " "); mainPlatformsParam != "" {
 		fmt.Println("> running the main builder docker image")
-		sh("docker run --rm -t -v $PWD:/app",
-			dockerMainBuilderImage, "-i", repoPath, "-p", q(mainPlatformsParam))
+		if err := docker("run --rm -t -v $PWD:/app", dockerMainBuilderImage, "-i", repoPath, "-p", q(mainPlatformsParam)); err != nil {
+			fatalMsg("The main builder docker image exited unexpectedly", err)
+		}
 	}
 
 	if armPlatformsParam := strings.Join(armPlatforms[:], " "); armPlatformsParam != "" {
 		fmt.Println("> running the ARM builder docker image")
-		sh("docker run --rm -t -v $PWD:/app",
-			dockerARMBuilderImage, "-i", repoPath, "-p", q(armPlatformsParam))
+		if err := docker("run --rm -t -v $PWD:/app", dockerARMBuilderImage, "-i", repoPath, "-p", q(armPlatformsParam)); err != nil {
+			fatalMsg("The ARM builder docker image exited unexpectedly", err)
+		}
 	}
 
 	if powerPCPlatformsParam := strings.Join(powerPCPlatforms[:], " "); powerPCPlatformsParam != "" {
 		fmt.Println("> running the PowerPC builder docker image")
-		sh("docker run --rm -t -v $PWD:/app",
-			dockerPowerPCBuilderImage, "-i", repoPath, "-p", q(powerPCPlatformsParam))
+		if err := docker("run --rm -t -v $PWD:/app", dockerPowerPCBuilderImage, "-i", repoPath, "-p", q(powerPCPlatformsParam)); err != nil {
+			fatalMsg("The PowerPC builder docker image exited unexpectedly", err)
+		}
 	}
 
 	if mipsPlatformsParam := strings.Join(mipsPlatforms[:], " "); mipsPlatformsParam != "" {
 		if version.Compare(goVersion, "1.6", ">=") {
 			fmt.Println("> running the MIPS builder docker image")
-			sh("docker run --rm -t -v $PWD:/app",
-				dockerMIPSBuilderImage, "-i", repoPath, "-p", q(mipsPlatformsParam))
+			if err := docker("run --rm -t -v $PWD:/app", dockerMIPSBuilderImage, "-i", repoPath, "-p", q(mipsPlatformsParam)); err != nil {
+				fatalMsg("The MIPS builder docker image exited unexpectedly", err)
+			}
 		} else {
 			warn(fmt.Errorf("MIPS architectures are only available with Go 1.6+"))
 		}

--- a/cmd/promu.go
+++ b/cmd/promu.go
@@ -31,6 +31,7 @@ import (
 var (
 	sh           = shell.Run
 	q            = shell.Quote
+	docker       = shell.Cmd("docker").ErrFn()
 	buildContext = build.Default
 	goos         = buildContext.GOOS
 	goarch       = buildContext.GOARCH

